plugins/snapcraft_travis/snapcraft_travis.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/python3
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2017 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import json
+import requests
+
+import errbot
+import git
+import github
+
+
+class SnapcraftTravis(errbot.BotPlugin):
+    """Handle Travis requests with the Travis API."""
+
+    def activate(self):
+        super().activate()
+
+    @errbot.arg_botcmd('pull_request_number', type=int)
+    def travis_snapurl(self, message, pull_request_number):
+        try:
+            build_id = self._get_build_id(pull_request_number)
+        except errbot.ValidationException:
+            return 'Could not find build with pull request #%i' % pull_request_number
+
+        jobs = self._travis_request('build/%s' % build_id, {})['jobs']
+        # get the id for the snap job
+        try:
+            job_id = jobs[3]['id']
+        except IndexError:
+            return 'Link not present in pull request #%i' % pull_request_number
+
+        log = self._travis_request('job/%s/log' % job_id, {})['content']
+        log = log.split('\n')
+        link = next((line for line in log if 'transfer.sh/' in line), 
+                    'Link not present in pull request #%i' % pull_request_number)
+        return link
+
+
+    def _travis_request(self, resource, params):        
```

Comment:
It would be better to have two functions called `_get_travis_job_log` and `_get_travis_requests`, or something like that. They call this one, and I would rename it to `_get_travis_api_resource`, or something similar.